includes/datatypes.php

```diff
@@ -848,3 +848,73 @@ function get_podcasting_language_options() {
 		)
 	);
 }
+
+/**
+ * Validate Podcast Taxonomy Fields.
+ *
+ * @param string $term     Term.
+ * @param string $taxonomy Taxonomy Name.
+ * @param array  $args     List of arguments.
+ *
+ * @return string
+ */
+function validate_taxonomy_fields( $term, $taxonomy, $args = [] ) {
+	// Bailout, if not the podcasts taxonomy.
+	if ( 'podcasting_podcasts' !== $taxonomy ) {
+		return $term;
+	}
+
+	$referer              = sanitize_text_field( $_POST['_wp_http_referer'] );
+	$is_onboarding_step_1 = false !== strpos( $referer, 'page=simple-podcasting-onboarding&step=1' );
+
+	if ( ! $is_onboarding_step_1 && empty( trim( $term ) ) ) {
+		return new \WP_Error( 'empty_term_name', __( 'A podcast name is required.', 'simple-podcasting' ) );
+	}
+
+	// The third argument was only introduced in the `pre_insert_term` filter in WP 6.1, so bail if it's empty.
+	if ( empty( $args ) ) {
+		return $term;
+	}
+
+	if ( $is_onboarding_step_1 ) {
+		$args['tag-name']              = sanitize_title( wp_unslash( $_POST['podcast-name'] ) );
+		$args['podcasting_category_1'] = sanitize_text_field( wp_unslash( $_POST['podcast-category'] ) );
+	}
+
+	// Require podcast name.
+	if ( empty( trim( $args['tag-name'] ) ) ) {
+		return new \WP_Error( 'empty_term_name', __( 'A podcast name is required.' ) );
+	}
+
+	// Require podcast author name only on term edit screen.
+	if ( empty( trim( $args['podcasting_talent_name'] ) ) ) {
+		return new \WP_Error( 'empty_term_talent_name', __( 'A podcast artist or author name is required.' ) );
+	}
+
+	// Require podcast description.
+	if ( empty( trim( $args['podcasting_summary'] ) ) ) {
+		return new \WP_Error( 'empty_term_summary', __( 'A podcast summary is required.' ) );
+	}
+
+	// Require podcast image.
+	if ( empty( trim( $args['podcasting_image'] ) ) ) {
+		return new \WP_Error( 'empty_term_cover_image', __( 'A podcast cover image is required.' ) );
```

Comment:
@mehul0810 Can we use the text domain for translatable strings?

Reply:
Sure, https://github.com/10up/simple-podcasting/pull/216/commits/558766c998d0742634c5bcb967389e2a814c5c47